### DISCUSSION TITLE
Question about the new inner radious of SeparationRayShape3D

### DISCUSSION THREAD

**Discussion by yosoyfreeman** [godot-jolt/godot-jolt#1005] [General]
I seen that in https://github.com/godot-jolt/godot-jolt/pull/1004 an inner radius was added to the ray in order to satisfy jolt. From the absolute ignorance, as i recognize i have no idea about this kind of resolution:

I remember fairly clearly that other separation shapes were not added because making them volumetric was complicated in someway. Seeing that now certain volume seems not only possible but a requirement, i was wondering if this could be used to implement something as a SeparationCylinder3D. The main reason for users not using the shape for what it was intended for (traversing uneven terrain/ Forced resolution in one direction) is that the lack of volume of the shape provides very little consistency. Making the already present radius workable on editor could do the trick too.

**Comment by mihe** [comment]
The inner radius added in #1004 has no relevance to the collision detection itself, as far as I'm aware, so there is still no volume to it. It's only used to determine whether CCD should be attempted for that body during that tick, which will no-op for that particular shape anyway, and was only made non-zero because of a Jolt assertion enforcing it as such.

**Reply by yosoyfreeman** [reply]
Thank you for the clarification Mihe!

Closing as resolved!
